### ISSUE TITLE
zkvm: clearer API distinction and documentation

### ISSUE THREAD

**Issue by vickiniu** [interstellar/slingshot#222]
In #218, the program calls `prove_program_predicate`, which should generally only be called indirectly when verifying a `Predicate::Program` variant. Looking at the zkvm codebase right now, it's a little confusing / unclear what functions should be used for building transactions (& which are internal to zkvm), especially since we expose many of them publicly.

Would be great to flesh out the [ZkVM API](https://github.com/interstellar/slingshot/blob/main/zkvm/docs/zkvm-api.md) doc with more traditional documentation of the functions, and also only expose as `pub` functions that should be used in tx-building.

**Comment by oleganza** [comment]
Yeah, i guess we can expand the [Predicates](https://github.com/interstellar/slingshot/blob/main/zkvm/docs/zkvm-api.md#predicates) section of the API guide on how to construct and interact with them (and add examples with Program::build/PredicateTree API)
